lib/internal/modules/cjs/loader.js

```diff
@@ -943,6 +946,10 @@ Module.prototype._compile = function(content, filename) {
 
 // Native extension for .js
 Module._extensions['.js'] = function(module, filename) {
+  const pkg = readPackageScope(filename);
+  if (pkg && pkg.type === 'module') {
+    throw new ERR_REQUIRE_ESM(filename);
```

Comment:
Should this error be behind the flag for now, or are we OK with throwing here without the `--experimental-modules` flag set?

Reply:
Just like we don't check the flag for loading from `.mjs`, it seems important to include unflagged as soon as possible to ensure we are reserving the compatibility space, unless you feel strongly about this?

Reply:
I think we want this unflagged or at the very least using a different flag. This is cleaning up an inconsistency with how `.mjs` and `.js` are handled so if throwing on `.mjs` isn't behind a flag, neither should this imo (and vice versa if I misremembered the current state :)).

Reply:
> it seems important to include unflagged as soon as possible to ensure we are reserving the compatibility space, unless you feel strongly about this?

Nah, I'm fine with it (hence the approve), just wanted to ask if it had been considered.